Stage0 built with clang-cl and linked with link.exe causes stack overflow on execution.

When using clang-cl.exe and link.exe to link, the resulting binary runs with 

I built this clang-cl from `088f33605 (grafted, HEAD, tag: llvmorg-15.0.6) [CodeGen][ARM] Fix ARMABIInfo::EmitVAAarg crash with empty record type variadic arg`

```
cmake -DCMAKE_CXX_COMPILER=clang -DCMAKE_C_COMPILER=clang -GNinja -DCMAKE_BUILD_TYPE=Debug -B build
ninja -C build
...
[92/129] Building CXX object CMakeFiles\jakt_stage0.dir\bootstrap\stage0\typechecker.cpp.obj
clang-cl: warning: unknown argument ignored in clang-cl: '-fdiagnostics-color=always' [-Wunknown-argument]
[94/129] Building jakt file selfhost/main.jakt with jakt_stage0
FAILED: jakt_stage1_tmp/main.cpp jakt_stage1_tmp/jakt__libc__io.cpp jakt_stage1_tmp/jakt__arguments.cpp <...SNIP...> F:/Source/jakt/build/jakt_stage1_tmp/utility.cpp
cmd.exe /C "cd /D F:\Source\jakt\build && F:\Source\jakt\build\bin\jakt_stage0.exe -S -T amd64-unknown-windows-unknown --binary-dir F:/Source/jakt/build/jakt_stage1_tmp --runtime-path F:/Source/jakt/runtime -I F:/Source/jakt/selfhost -IF:/Source/jakt/runtime F:/Source/jakt/selfhost/main.jakt"
ninja: build stopped: subcommand failed.
```

running the offending command in isolation, which is supposed to create cpp files for all of the jakt modules in the selfhost compiler crashes:

```
PS F:\Source\jakt> F:\Source\jakt\build\bin\jakt_stage0.exe -S -T amd64-unknown-windows-unknown --binary-dir F:/Source/jakt/build/jakt_stage1_tmp --runtime-path F:/Source/jakt/runtime -I F:/Source/jakt/selfhost -IF:/Source/jakt/runtime F:/Source/jakt/selfhost/main.jakt
PS F:\Source\jakt> $LASTEXITCODE
-1073741571
```

The exit code is 0xC00000FD, which is a stack overflow.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,6 +76,9 @@ target_include_directories(jakt_stage0
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/runtime>
 )
 apply_output_rules(jakt_stage0)
+if (MSVC)
+    target_link_options(jakt_stage0 PRIVATE LINKER:/STACK:0x800000)
+endif()
 
 set(SELFHOST_SOURCES
   selfhost/build.jakt
@@ -148,6 +151,9 @@ target_include_directories(jakt_stage1
 )
 add_executable(Jakt::jakt_stage1 ALIAS jakt_stage1)
 apply_output_rules(jakt_stage1)
+if (MSVC)
+    target_link_options(jakt_stage1 PRIVATE LINKER:/STACK:0x800000)
+endif()
 
 if (FINAL_STAGE GREATER_EQUAL 2)
   add_jakt_executable(jakt_stage2
@@ -165,6 +171,9 @@ if (FINAL_STAGE GREATER_EQUAL 2)
   )
   add_executable(Jakt::jakt_stage2 ALIAS jakt_stage2)
   apply_output_rules(jakt_stage2)
+  if (MSVC)
+    target_link_options(jakt_stage2 PRIVATE LINKER:/STACK:0x800000)
+  endif()
 endif()
 
 # Link runtime into build directory(ies) for relative pathing usage

--- a/cmake/jakt-executable.cmake
+++ b/cmake/jakt-executable.cmake
@@ -25,7 +25,6 @@ function(add_jakt_compiler_flags target)
     -Wno-deprecated-declarations
     -Wno-unknown-warning-option
     -Wno-unused-command-line-argument
-    -fdiagnostics-color=always
     # Silence warning about `no_unique_address`;
     # It does not apply on windows, and clang-cl just warns about it.
     -Wno-unknown-attributes
@@ -34,7 +33,7 @@ function(add_jakt_compiler_flags target)
     # For clang-cl, which shows up to CMake as MSVC and accepts both kinds of arguments
     target_compile_options("${target}" PRIVATE /permissive- /utf-8 /EHsc-)
   else()
-    target_compile_options("${target}" PRIVATE -fno-exceptions)
+    target_compile_options("${target}" PRIVATE -fno-exceptions -fdiagnostics-color=always)
   endif()
   if (CYGWIN OR MSYS)
     target_compile_options("${target}" PRIVATE -Wa,-mbig-obj)

